src/services/publisher_service/helpers.rs

```diff
@@ -156,12 +153,15 @@ pub async fn pick_subscriber_notification_for_processing(
 }
 
 #[instrument(skip(postgres, metrics))]
-pub async fn update_message_processing_status(
+pub async fn update_message_processing_status<'e, E>(
     notification: Uuid,
     status: SubscriberNotificationStatus,
-    postgres: &PgPool,
+    postgres: E,
```

Comment:
Maybe we could use `PgExecutor` instead to simplify? Looks like fulfills similar purpose as `PgPool`

```suggestion
    postgres: impl PgExecutor<'e>,
```

Reply:
If it looks more readable, I don't mind to change it ;)